### PR TITLE
Tests fix due to geth release changes

### LINKED ISSUE #6364: Tests fix due to geth release changes
## Description

Fixes of failing geth http tests:
https://github.com/web3/web3.js/actions/runs/5835730499/job/15828038229?pr=6346#step:5:282
https://github.com/web3/web3.js/actions/runs/5835730499/job/15828038229?pr=6346#step:5:301
https://github.com/web3/web3.js/actions/runs/5835730499/job/15828038229?pr=6346#step:5:319
https://github.com/web3/web3.js/actions/runs/5835730499/job/15828038229?pr=6346#step:5:1153

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### PR DESCRIPTION
**Pull request by jdevcs**
## Description

Fixes of failing geth http tests:
https://github.com/web3/web3.js/actions/runs/5835730499/job/15828038229?pr=6346#step:5:282
https://github.com/web3/web3.js/actions/runs/5835730499/job/15828038229?pr=6346#step:5:301
https://github.com/web3/web3.js/actions/runs/5835730499/job/15828038229?pr=6346#step:5:319
https://github.com/web3/web3.js/actions/runs/5835730499/job/15828038229?pr=6346#step:5:1153

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Review by avkos — CHANGES_REQUESTED** [review]
Thanks for these fixes. 
Unfortunately, we still have some tests which do not pass.

**Comment by jdevcs** [comment]
> Thanks for these fixes. Unfortunately, we still have some tests which do not pass.

I'll fix ws and ipc in another PR, after merging it? as ws and ipc are not due to same issue I think.

**Review by luu-alex — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -191,28 +191,6 @@ jobs:
       -   run: yarn test:e2e:${{ matrix.backend }}:${{ matrix.mode }}
           shell: bash
 
-  e2e-sync:
```
Comment (jdevcs) [inline .github/workflows/build.yml:194]: Starting from 1.12.0 geth doesn't support ethash and stops with : "`Fatal: Failed to register the Ethereum service: ethash is only supported as a historical component of already merged networks`" so sync test scripts are removed

Comment (avkos) [reply .github/workflows/build.yml:194]: OMG. do we need somehow to adapt our sync test? Do we need to create an issue for this?

Comment (jdevcs) [reply .github/workflows/build.yml:194]: As Eth mainnet is on POS so we dnt need to invoke sync tests, but let these be in tests so any one using web3.js on other NWs with POW, the functionality will be there and we can invoke tests manually if want to debug or see.

**Review by avkos — APPROVED** [review]

**Pull request merged**